tasks/main.yml

```diff
@@ -17,26 +12,328 @@
+
+    - name: Change the edition of MSSQL Server  # noqa 102
+      command: /opt/mssql/bin/mssql-conf set-edition
+      environment:
+        MSSQL_PID: "{{ mssql_edition }}"
+      register: __mssql_conf_set_edition
+      changed_when: '"The new edition is" in __mssql_conf_set_edition.stdout'
+      when: >
+        ((__mssql_errorlog_custom_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_custom_content.stdout | lower)) or
+        ((__mssql_errorlog_default_content.skipped is not defined) and
+        ("{{ mssql_edition | lower }} edition" not in
+        __mssql_errorlog_default_content.stdout | lower))
+      # Skipping rule because {{}} are required around vars here
+      notify: Restart the mssql-server service
+
+- name: Set the IP address setting for MSSQL Server
+  when:
+    - __mssql_conf_setup.skipped is defined
+    - mssql_ip_address is defined
+  block:
+    - name: Check if the IP address setting is defined
+      command: /opt/mssql/bin/mssql-conf get network ipaddress
+      ignore_errors: true
+      changed_when: false
+      register: __mssql_conf_get_ip_address
+
+    - name: Change the IP address setting
+      command: >
+        /opt/mssql/bin/mssql-conf set network.ipaddress {{ mssql_ip_address }}
+      when: >
+        ("No setting for the given" in __mssql_conf_get_ip_address.stdout) or
+        (__mssql_conf_get_ip_address.stdout | regex_search('[^: ]*$')
+        not in mssql_ip_address)
+      notify: Restart the mssql-server service
+
+- name: Set the new TCP port for MSSQL Server
+  command: "/opt/mssql/bin/mssql-conf set network.tcpport {{ mssql_tcp_port }}"
+  register: __mssql_conf_set_tcp_port
+  ignore_errors: true
+  changed_when: >
+    "SQL Server needs to be restarted in order to apply this setting."
+    in __mssql_conf_set_tcp_port.stdout
+  notify: Restart the mssql-server service
+  when:
+    - __mssql_conf_setup.skipped is defined
+    - mssql_tcp_port is defined
+
+- name: Input the {{ mssql_input_sql_file }} file to MSSQL Server
+  when: mssql_input_sql_file is defined
+  block:
+    - name: Create a tempfile on the host
+      tempfile:
+        state: file
+      register: __mssql_sql_tempfile
+
+    - name: Copy the {{ mssql_input_sql_file }} file to the host
+      copy:
+        src: "{{ mssql_input_sql_file }}"
+        dest: "{{ __mssql_sql_tempfile.path }}"
+        mode: preserve
+
+    - name: Ensure that the mssql-server service is started
+      service:
+        name: mssql-server
+        state: started
+
+    - name: Input the {{ mssql_input_sql_file }} file with the sqlcmd command
+      command: |
+        /opt/mssql-tools/bin/sqlcmd -U sa -P "{{ mssql_password }}"
+        -i {{ __mssql_sql_tempfile.path }}
+      register: __mssql_sqlcmd
+
+    - name: Print the output of the sqlcmd command
+      debug:
+        var: __mssql_sqlcmd.stdout_lines
+      when: mssql_input_sql_file is defined
```

Comment:
I don't think you need this `when` since the entire block is covered by the same condition

Reply:
True, removed.

Reply:
True,done